lib/api/src/backend/sys/entities/exception.rs

```diff
@@ -2,87 +2,101 @@
 use std::any::Any;
 
 use wasmer_types::{TagType, Type};
-use wasmer_vm::StoreHandle;
+use wasmer_vm::{StoreHandle, StoreId};
 
 use crate::{
-    AsStoreMut, AsStoreRef, Tag, Value,
-    sys::vm::{VMException, VMExceptionRef},
+    AsStoreMut, AsStoreRef, BackendTag, Tag, Value,
+    sys::vm::{VMExceptionObj, VMExceptionRef},
 };
 
 #[derive(Debug, Clone, PartialEq, Eq)]
-/// A WebAssembly `exception` in the `sys` runtime.
+/// A WebAssembly `exnref` in the `sys` runtime.
 pub(crate) struct Exception {
-    pub(crate) handle: VMException,
+    exnref: wasmer_vm::VMExceptionRef,
 }
 
-unsafe impl Send for Exception {}
-unsafe impl Sync for Exception {}
-
 impl Exception {
-    /// Create a new [`Exception`].
-    pub fn new(store: &mut impl AsStoreMut, tag: Tag, payload: &[Value]) -> Self {
-        todo!()
-    }
-}
+    /// Creates a new exception with the given tag and payload, and also creates
+    /// a reference to it, returning the reference.
+    #[allow(irrefutable_let_patterns)]
+    pub fn new(store: &mut impl AsStoreMut, tag: &Tag, payload: &[Value]) -> Self {
+        if !tag.is_from_store(store) {
+            panic!("cannot create Exception with Tag from another Store");
+        }
 
-#[derive(Debug, Clone)]
-#[repr(transparent)]
-/// A WebAssembly `extern ref` in the `sys` runtime.
-pub(crate) struct ExceptionRef {
-    handle: StoreHandle<wasmer_vm::VMExceptionObj>,
-}
+        let BackendTag::Sys(tag) = &tag.0 else {
+            panic!("cannot create Exception with Tag from another backend");
+        };
```

Comment:
Move this logic one level up. This 'SysException" type can receive the Sys type inside of the `BackendTag::Sys instead of the BackendTag itself.